### PR TITLE
Shopify Watch/Deploy Extra

### PR DESCRIPTION
**Pull request by solomonhawk**
Hey @greypants, I'm not _certain_ this belongs in an extra but it seems fairly suitable. I've found this to be helpful recently.

### CODE REVIEW

**Comment by greypants** [comment]
👏  Nice!

File: extras/shopify/gulpfile.js/tasks/shopify.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    .pipe(notify({
+      message: '<%= file.relative %> uploaded.'
+    }));
+});
+
+gulp.task('shopifydeploy', function() {
+  return gulp.src('./+(assets|config|layout|snippets|templates|locales)/**')
+    .pipe(plumber({
+      errorHandler: notify.onError("Error: <%= error.message %>")
+    }))
```
Comment (greypants) [inline extras/shopify/gulpfile.js/tasks/shopify.js]: you may be able to use `lib/handleErrors` instead and drop the gulp plumber dependency. OR I should kill `lib/handleErrors` and use gulp plumber. There's an open issue about it: https://github.com/vigetlabs/gulp-starter/issues/171

Opinions?

Comment (solomonhawk) [inline extras/shopify/gulpfile.js/tasks/shopify.js]: I think changing this to just use `lib/handleErrors` is fine.

Comment (greypants) [inline extras/shopify/gulpfile.js/tasks/shopify.js]: Make it so 👉

**Comment by greypants** [comment]
@solomonhawk ping!

**Comment by solomonhawk** [comment]
@greypants How does this look? (can you just verify that I've used `handleErrors` correctly in 09c8cf0 ?)

File: extras/shopify/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "main": "gulpfile.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Solomon Hawk <solomon.hawk@viget.com> (http://viget.com/)",
+  "license": "ISC",
+  "dependencies": {
+    "gulp": "3.9.1",
+    "gulp-notify": "2.2.0",
+    "gulp-plumber": "1.1.0",
```
Comment (greypants) [inline extras/shopify/package.json]: Think you can remove this now.

File: extras/shopify/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "gulp-shopify-upload": "2.0.0",
+    "gulp-watch": "4.3.5"
+  }
+}
```
Comment (greypants) [inline extras/shopify/package.json]: should this only include dependencies  not already included in gulp starter's package.json? Would users just copy these into that one, or would you have two separate package.jsons?

**Comment by greypants** [comment]
Looks pretty good! Could you add a `config.json` with the paths configured for the shopify folder structure?

File: extras/shopify/package.json (hunk 3 above)
Comment (solomonhawk) [inline extras/shopify/package.json]: In the nested README for this dir there is a note about `package.json`. 

```
#### `package.json`

Included in this example is a `package.json` with dependencies needed to use these gulp tasks. You'll need to either install them by name or copy the dependencies over and `npm install`.
```

Did you have something else in mind?

Comment (greypants) [inline extras/shopify/package.json]: Cool. I was just thinking you'd list the *additional* dependencies, so in this case, just `gulp-shopify-upload`. But maybe it's better just to list them all?

**Pull request merged**